src/main/java/seedu/address/model/timeslots/UniqueTimeSlotList.java

```diff
@@ -0,0 +1,156 @@
+     * @param replacement UniqueTimeSlotList we want to set current List to
+     */
+    public void setTimeslots(UniqueTimeSlotList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code timeslots}.
+     * {@code timeslots} must not contain duplicate Timeslots.
+     */
+    public void setTimeslots(List<Timeslots> timeSlotsList) {
+        requireAllNonNull(timeSlotsList);
+        if (!timeSlotsAreUnique(timeSlotsList)) {
+            throw new DuplicateTimeslotException();
+        }
+
+        internalList.setAll(timeSlotsList);
+    }
+
+    /**
+     * Adds to the contents of this list with {@code timeslots}.
+     * {@code timeslots} must not contain duplicate Timeslots.
+     */
+    public void addAll(List<Timeslots> timeslotsList) {
+        requireAllNonNull(timeslotsList);
+        if (!timeSlotsAreUnique(timeslotsList)) {
+            throw new DuplicateTimeslotException();
+        }
+
+        internalList.addAll(timeslotsList);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Timeslots> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    public void setTimeslotsList(List<Timeslots> timeslotsList) {
+        requireNonNull(timeslotsList);
+        if (!timeSlotsAreUnique(timeslotsList)) {
+            throw new DuplicateTimeslotException();
+        }
+    }
+
+    @Override
+    public Iterator<Timeslots> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof UniqueTimeSlotList)) {
+            return false;
+        }
+
+        UniqueTimeSlotList otherUniqueTimeSlotList = (UniqueTimeSlotList) other;
+        return internalList.equals(otherUniqueTimeSlotList.internalList);
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    /**
+     * Returns the size of the UniqueTimeSlotList
+     * @return Size of the internalList instance
+     */
+    public int size() {
+        return internalList.size();
+    }
+
+    /**
+     * Returns true if {@code timeslots} contains only unique Timeslots.
+     */
+    private boolean timeSlotsAreUnique(List<Timeslots> timeslotsList) {
+        for (int i = 0; i < timeslotsList.size() - 1; i++) {
+            for (int j = i + 1; j < timeslotsList.size(); j++) {
+                if (timeslotsList.get(i).isSameTimeSlot(timeslotsList.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
maybe you can apply SLAP here so it won't be so nested

Reply:
Great spot, replace deep nested code by using a Set instead